src/compiler/emitter.ts

```diff
@@ -4502,7 +4518,212 @@ module ts {
                 });
             }
 
-            function emitClassDeclaration(node: ClassDeclaration) {
+            function emitMemberFunctionsForES6AndHigher(node: ClassDeclaration) {
+                forEach(node.members, member => {
+                    if (member.kind === SyntaxKind.MethodDeclaration || node.kind === SyntaxKind.MethodSignature) {
+                        if (!(<MethodDeclaration>member).body) {
+                            return emitPinnedOrTripleSlashComments(member);
+                        }
+
+                        writeLine();
+                        emitLeadingComments(member);
+                        emitStart(member);
+                        if (member.flags & NodeFlags.Static) {
+                            write("static ");
+                        }
+                        emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
+                        emitSignatureAndBody(<MethodDeclaration>member);
+                        emitEnd(member);
+                        emitTrailingComments(member);
+                    }
+                    else if (member.kind === SyntaxKind.GetAccessor || member.kind === SyntaxKind.SetAccessor) {
+                        var accessors = getAllAccessorDeclarations(node.members, <AccessorDeclaration>member);

```

Comment:
I'm not convinced you need to get all the accessors and emit them together. I would just have one emit method, for methods and accessors, and just emit the word "get" or "set" if you need to, the same way you emit "static".